src/librustc_mir/hair/cx/expr.rs

```diff
@@ -320,6 +320,8 @@ impl<'tcx> Mirror<'tcx> for &'tcx hir::Expr {
                                   name: Field::new(index.node as usize) },
             hir::ExprCast(ref source, _) =>
                 ExprKind::Cast { source: source.to_ref() },
+            hir::ExprType(ref source, _) =>
+                return source.make_mirror(cx),
```

Comment:
This doesn't seem right. I think we probably want to convert this to a `ExprKind::Cast`.